### ISSUE TITLE
Assistant fails to return response if tool returns large string

### ISSUE THREAD

**Issue by c9482** [phidatahq/phidata#134]
I have created the following program to test phidata tools capabilities. I noticed that when my tool returns a larger string, the programs invokes the tool in an infinite loop (for example string 4K in size). I also get the following warning:

> WARNING  Could not parse tool calls from response:  {                                                                                                  
>          "tool\_calls": [                                                                                                                              
>          {                                                                                                                                             
>          "name": "xvpinfo",                                                                                                                            
>          "arguments": {                                                                                                                                
>          "street": "Swan Lake",                                                                                                                       
>          "year": "2024"                                                                                                                                
>          }                                                                                                                                             
>          }                                                                                                                                             
>          ]                                                                                                                                             
>          }  

When the tool returns smaller string (for example 500 bytes in size), everything works as expected. 

Is there anything to tweak to make the program work with larger strings returned by the tool? Many thanks in advance.

```
import logging
from phi.assistant import Assistant
from phi.llm.ollama import Ollama
import requests
import sys
from phi.tools import Toolkit

temp = 0.3
models = ["mistral", "llama2", "mixtral"]
model = models[0]

class XvpTool(Toolkit):
    def __init__(Toolkit):
        super().__init__()

    def xvpinfo(self, street: str, year: str) -> str:
        """Retrieves list of residents living on a street in specific year.
        Args:
            street (str): Full or part of street name (e.g., 'SWAN LAKE', '930 MAIN').
            year (str): year (e.g., '2024').            
        Returns:
            str: A list of residents and their full address.
        """
        url = f"https://secret.com/api/search/address/{street}/{year}/"
        headers = {
            "API-Key": "******",
            "X-Version": "2.0"
        }

        try:
            response = requests.get(url, headers=headers)

            # Check if the request was successful (status code 200)
            if response.status_code == 200:
                result_set = response.json()

                if not result_set:
                    return "None"
                result_string = "\r\n".join([f"{entry.get('name', '')} {entry.get('locNameFull', '')}" if 'locNameFull' in entry else f"{entry.get('name', '')}" for entry in result_set])
                return result_string
            else:
                print(f"Error: {response.status_code} - {response.text}")
                return None

        except Exception as e:
            print(f"Error: {e}")
            return None

assistant = Assistant(
    llm=Ollama(model=model, options={"temperature": temp}),
    description="You are a helpful Assistant to retrieve list of residents living on specific street using tools", 
    instructions=[
            "Given a street name and year, retrieve a list of residents and their address sorted by owner name.",
            "At the end of the response, please include the number of residents found.",
            "At the end of the response, please provide resident count by city.", 
            "If no residents are found, please return 'None Found'."
    ],
    tools=[XvpTool().xvpinfo], 
    show_tool_calls=True,
    debug_mode=True,
    prevent_hallucinations=True,
)

assistant.print_response(f"Who lives at {street} in year {year}?", markdown=False)


```

**Comment by ashpreetbedi** [comment]
@c9482 this is really cool and thanks for reporting, could you try this with gpt4? I think this might be a limitation of local models.

As they're smaller & not as competent, they tend to have errors like:
- invoking tools in an infinite loop.
- not sending the correct response for the tool call to be parsed.

These are both issues you're seeing and just the limitation of local/smaller models not being as good sadly. I'm confident by the end of this year the local models will catch up.

In the meanwhile please can you test with gpt4 and let me know if you still see these issues?

**Comment by c9482** [comment]
@ashpreetbedi thanks for the suggestion. Just tried it with OpenAI and it works fine. 

I wonder if it's a limitation of Ollama. 

Or perhaps I need to pass an option to the llm parameter but I am not sure which one?

**Comment by ashpreetbedi** [comment]
@c9482 yes sadly its a limitation of smaller models. The smaller/OSS models just aren't as good at advanced features like function calling. But from what im hearing they're catching up fast and maybe in a month or 2 the reliability should improve greatly

**Comment by ashpreetbedi** [comment]
Going to close, but please re-open if needed :) Also happy to chat on our discord about this: https://discord.com/invite/4MtYHHrgA8

**Comment by c9482** [comment]
@ashpreetbedi thanks again. Actually, I modified the code to include the limit parameter and it seems to be working fine now with the same criteria that would not work before. I tried it with limit as high as 50 and mistral. I am a Python and AI novice, so I don't understand why this would change anything. 

`result_string = "\r\n".join([f"{entry.get('name', '')} {entry.get('locNameFull', '')}" if 'locNameFull' in entry else f"{entry.get('name', '')}" for entry in result_set[:limit]])`
